### PR TITLE
CLN: DataFrameGroupBy._cython_agg_general

### PR DESCRIPTION
**Pull request by topper-123**
Some small cleanups that make the code easier to read IMO.

### CODE REVIEW

File: pandas/core/groupby/generic.py
Comment (jbrockmendel) [inline pandas/core/groupby/generic.py]: if Block is only used for annotations, can you put it in a `if TYPE_CHECKING`

File: pandas/core/groupby/generic.py
Comment (jbrockmendel) [inline pandas/core/groupby/generic.py]: while you're at it, can you rename `blk` to `blks` or something else clearly plural

Comment (topper-123) [reply pandas/core/groupby/generic.py]: Ok

File: pandas/core/groupby/generic.py
Comment (topper-123) [reply pandas/core/groupby/generic.py]: Ok.

File: pandas/core/groupby/generic.py
Comment (jbrockmendel) [inline pandas/core/groupby/generic.py]: @WillAyd is getting Block/BlockManager stuff out of here still something you're working on?

**Comment by jbrockmendel** [comment]
LGTM

**Comment by jbrockmendel** [comment]
cc @WillAyd for a second pair of eyes

File: pandas/core/groupby/generic.py
Comment (WillAyd) [inline pandas/core/groupby/generic.py]: I would actually prefer to keep this named `_wrap_agged_blocks` since it offers the same functionality as the rest of the `_wrap_*` functions in groupby, albeit with different input types

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/generic.py
Comment (topper-123) [reply pandas/core/groupby/generic.py]: That's a fair point, I wasn't aware of this systematic naming, I'll change it back. With the added typing we now see better that a frame is returned

File: pandas/core/groupby/generic.py
Comment (topper-123) [inline pandas/core/groupby/generic.py]: I've inverted the argument order. I think it makes more sense to have the blocks come first, like in ``BlockManager``.

File: pandas/core/groupby/generic.py
Comment (topper-123) [inline pandas/core/groupby/generic.py]: Just some minor cleanups above: pluralizing names + use list comprehension instead of map.

Comment (jbrockmendel) [reply pandas/core/groupby/generic.py]: i tend to prefer these too

Comment (WillAyd) [reply pandas/core/groupby/generic.py]: Can this just be a generator expression or does that fail? I see this is equivalent to existing code but maybe adds unnecessary overhead

Comment (topper-123) [reply pandas/core/groupby/generic.py]: It would work, but wouldn't make any difference because it's passed into the BlockManager where it's stored.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
lgtm @jreback if you care to look

**Comment by jreback** [comment]
thanks!

**Pull request merged**
